Source/Reporters/JunitXMLReporter.jsl

```diff
@@ -0,0 +1,247 @@
+		Values will be associative arrays with test cases as keys.
+		The values of those associative arrays will be failure messages with "failure" as the key.
+
+    Example:
+        This is an example
+        ---JSL---
+		xml_results = ["Example Tests" => 
+					    ["Test1" => 
+						  ["failure" =>
+							"Expected: add_nums(1, 1) equal to 3
+							But: was 2"
+						  ]
+						]
+					  ];
+		show_xml_report(xml_results);			
+
+		Expected Return:
+		<testsuites failures="1" id="20191011_175324" tests="1">
+			<testsuite failures="1" name="Example Tests" tests="1">
+				<testcase name="Test1">
+					<failure type="failure">
+						Expected: add_nums(1, 1) equal to 3
+						But: was 2
+					</failure>
+				</testcase>
+			</testsuite>
+		</testsuites>
+        ---------
+*/
+	show_xml_report = Method({xml_results},
+		start_line = "<?xml version=\!"1.0\!" encoding=\!"UTF-8\!" ?>\!n";
+		
+		content = "";
+		test_suites = xml_results << get keys;
+		test_cases_set = xml_results << get values;
+		For(i=1, i<=NItems(test_suites), i++,
+		
+
+			test_cases = test_cases_set[i];
+			test_case_names = test_cases << get keys;
+			test_case_info = test_cases << get values;
+			test_case_xml = "";
+			For(k=1, k<=NItems(test_case_names), k++,
+
+				info_types = test_case_info[k] << get keys;
+				
+				test_case_content = "";
+				For(l=1, l<=NItems(test_case_info[k]), l++,
+					info_type = info_types[l];
+					props = Associative Array();
+					props["type"] = info_type;
+					failure_content = this:generate_xml_tag("failure", test_case_info[k][info_type], props);
+
+					test_case_content ||= failure_content;
+
+				);
+				props = Associative Array();
+				props["name"] = test_case_names[k];
+				test_case_xml ||= this:generate_xml_tag("testcase", trim(test_case_content), props);
+				
+			);
+			num_tests = NItems(test_case_names);
+			props = Associative Array();
+			props["name"] = test_suites[i];
+			props["tests"] = Char(num_tests);
+
+			// find test results that are non empty
+			num_successes = Nitems(loc(test_case_info, [=>]));
+			props["failures"] = Char(num_tests-num_successes);
+			test_suite_tag = this:generate_xml_tag("testsuite", trim(test_case_xml), props);
+			content ||= test_suite_tag;
+		);
+		
+		props = Associative Array();
+		date = today();
+		props["id"] = format(date, "yyyymmdd" ) || "_" || Char(Hour(date)) || Char(Minute(date)) || Char(Second(date));
+		props["tests"] = Char(Nitems(this:successes) + NItems(this:failures) + NItems(this:unexpected throws));
+		props["failures"] = Char(NItems(this:failures) + NItems(this:unexpected throws));
+
+		xml_result = start_line || this:generate_xml_tag("testsuites", trim(content), props);;
+		Return(xml_result);
+	);
+	
+	// Function: show report
+	//
+	// Dump a report to the log detailing all the successes and
+	// failures for each test in JunitXML format shown in the comments at the top.
+	show report = Method( {},
```

Comment:
Since this `show report` method has a lot of logic to create the result string, could we change this to `get report` that returns the string and add `show report` that just calls `Write(this:get report)`? This would allow you to do other things with the report, like write it to a file.

Reply:
This was resolved by @emccorkle's commit.